src/pages/top-up/Swap.tsx

```diff
@@ -127,6 +134,27 @@ export function Swap({ header }: Props): ReactElement {
     }
   }
 
+  async function performSwapWithChecks(daiToSwap: DaiToken) {
```

Comment:
Sooo I am not really content with how the error handling is done. I would suggest removing all the `enqueueSnackbar` cases here and leaving it to the top-level `try/catch` in the `onSwap()` function to do the error presentation.

I would also suggest creating a custom Error something like `SwapError` that would have a property  `snackMessage` or similar, where the custom error messages would be placed.

Then I would suggest a helper function called something like `wrapErrorWithMessage<T>(promise: Promise<T>, errorMessage: string): Promise<T>` that would catch an error and would wrap it with the `SwapError` with your snack message and rethrow. In this way, you avoid the `.catch()` handling.

Reply:
OK I generally agree, what I want to achieve here is to have the full raw error logged to console, and a nicer and clear error message on the screen. I'll implement both with your approach.

Reply:
Nice 👍